### PR TITLE
First draft of hard coded priority rules

### PR DESCRIPTION
**Pull request by ojanvafai**
Eventually, we'll build UI to configure this and this will just be a single available scheme of multiple.

### CODE REVIEW

File: src/ui/App.tsx [hunk 1]
```diff
@@ -27,6 +27,105 @@ enum LoadState {
   loaded,
 }
 
+// TODO: Add setaside
```
Comment (tdresser) [inline src/ui/App.tsx:30]: Might be premature, but should we keep this out of App.tsx?

File: src/ui/App.tsx [hunk 2]
```diff
@@ -27,6 +27,105 @@ enum LoadState {
...
+  hidden,
+}
+
+enum BuiltInAction {
+  archive,
+  moveToPriority,
+}
+
+// If the action value is a string, it's the name of the priority to move to.
+type Action = BuiltInAction | string;
```
Comment (tdresser) [inline src/ui/App.tsx:42]: We'll be representing priorities as gmail labels. Should we wrap this in something making it easier to manager mapping from an action to the associated gmail label?

File: src/ui/App.tsx [hunk 3]
```diff
@@ -27,6 +27,105 @@ enum LoadState {
...
+  {
+    name: 'Stuck',
+    visibility: PriorityVisibility.visible,
+  },
+  {
+    name: 'Stuck/%epoch%',
+    visibility: PriorityVisibility.hidden,
+    triggers: [
+      { type: TriggerType.newMessage, action: 'Stuck' },
+      { type: TriggerType.deadline, action: 'Stuck', allowCustomDeadlines: true },
```
Comment (tdresser) [inline src/ui/App.tsx:92]: Is the "action" here correct?

File: src/ui/App.tsx [hunk 4]
```diff
@@ -27,6 +27,105 @@ enum LoadState {
...
+    ],
+  },
+  {
+    name: 'mk/priority/Empty-Daily',
+    visibility: PriorityVisibility.hidden,
```
Comment (tdresser) [inline src/ui/App.tsx:97]: Why is this hidden?

**Review by tdresser — COMMENTED** [review]
Looking pretty nice!

Mostly idle thoughts that are probably premature.

File: src/ui/App.tsx (hunk 2 above)
Comment (ojanvafai) [reply src/ui/App.tsx:42]: I was thinking the string would be the gmail label name. Is there other metadata we'd also want to track?

File: src/ui/App.tsx (hunk 3 above)
Comment (ojanvafai) [reply src/ui/App.tsx:92]: Yeah, see above that Stuck without the trailing epoch is PriorityVisibility.visible. So, this is making it visible in the inbox again.

Although, now that I write that, I realize that I'm merging two concepts into visible. There are visible-and-triaged and needs-triage. Those need to be separate visibility values.

File: src/ui/App.tsx (hunk 4 above)
Comment (ojanvafai) [reply src/ui/App.tsx:97]: Typo. :)

File: src/ui/App.tsx (hunk 1 above)
Comment (ojanvafai) [reply src/ui/App.tsx:30]: yup

File: src/ui/App.tsx (hunk 2 above)
Comment (tdresser) [reply src/ui/App.tsx:42]: I was thinking we might want to directly include a "Label" object.
https://github.com/ojanvafai/tskod/blob/6f706f62b328caa920ec8c17e4b3a8c15ecf165e/src/Labels.tsx#L19

Not positive that makes sense though. Managing lazy instantiation might be a pain.

**Pull request closed without merging**